### ISSUE TITLE
Galata stuck when loading for JupyterLab to be ready on Firefox

### ISSUE THREAD

**Issue by jtpio** [jupyterlab/jupyterlab#15093]
## Description

UI tests time out when using Galata with the Firefox browser. This happens when loading the default JupyterLab page, and might be related to the `waitForApplication` logic.

## Reproduce

- Create a new extension from the extension template: https://github.com/jupyterlab/extension-template
- Follow the instructions to setup the UI tests and install playwright
- Run `jlpm test --browser firefox`
- The tests time out

**chromium**

https://github.com/jupyterlab/jupyterlab/assets/591645/ba15fcba-2c61-42e2-b4b2-b9e17ff4a610

**firefox**

https://github.com/jupyterlab/jupyterlab/assets/591645/955e6186-70f9-4a31-beb3-07d9d36a023f

## Expected behavior

The tests should run fine like on Chromium.

## Context

This was first noticed in https://github.com/jupyterlite/jupyterlite/pull/1019 (which tests on both `chromium` and `firefox`), but it easily reproducible with the extension template.

This is related to https://github.com/jupyterlab/jupyterlab/issues/15088, which can help catch such issues.

**Comment by jtpio** [comment]
Tagging as regression as this was not an issue with the `3.5.x` packages.

**Comment by krassowski** [comment]
Can we close this? We run tests against Firefox on `main` so I think this would no longer be a problem?

**Comment by jtpio** [comment]
Yeah, it looks good now. I just tried again with the same instructions as mentioned above with the extension template. Now it seems to be passing fine:

<img width="1624" height="88" alt="Image" src="https://github.com/user-attachments/assets/fa711165-cd20-4c19-95ff-cdb60e2623fd" />

Thanks!
